src/dialog/Modal.jsx

```diff
@@ -76,7 +80,13 @@ export function Modal({
           size && `modal-${size}`,
         ])}
         role="document"
-        onClick={stopPropagation}
+        onClick={(e) => {
+          e.forceStopPropagation = true;
+
+          if (stopPropagation) {
+            stopPropagationEvent(e);
+          }
```

Comment:
Quando a Modal precisar propagar um evento de clique realizado dentro dela (por exemplo, para avisar o FormDropdown que precisa fechar suas opções), é necessário forçar um stopPropagation para que o onClick da linha 65, ao receber o evento propagado, entenda que não é para executar o fechamento da modal caso forceStopPropagation = true. 

Caso o evento da linha 65 chegue por outro lugar (normalmente quando o usuário clica na área externa da modal), ele será executado normalmente. Caso chegue pelo clique dentro da modal, não será executado, impedindo seu fechamento.